torch/csrc/distributed/rpc/process_group_agent.cpp

```diff
@@ -234,18 +234,19 @@ void ProcessGroupAgent::sync() {
 void ProcessGroupAgent::start() {
   {
     std::lock_guard<std::mutex> futureLock{futureMutex_};
```

Comment:
Why do we need this lock here? Does the following comment also apply here? Are we worrying about two threads, one calling `start` and another calling `shutdown`? I am unclear of how this could go wrong. 

```c++
      // If during evaluating the predicate, `::shutdown()` is called, then
      // the predicate missed the notification before it started waiting
      // on the cond var.
```

Reply:
Made all the requested changes except this one - running some tests internally, and will update with this change shortly!